### ISSUE TITLE
Stopped Reason field not populated on "Failed to Start"

### ISSUE THREAD

**Issue by PiedPieper** [aws/amazon-ecs-agent#1654]
### Summary
I am trying to run a basic Docker container on ECS, and I keep getting the error "STOPPED(Task failed to start)

### Description
In the documentation (https://docs.aws.amazon.com/AmazonECS/latest/developerguide/stopped-task-errors.html) it says to, "In the Details section, inspect the Stopped reason field to see the reason that the task was stopped".

There is no "Stopped reason" field.

Here is what my Details section looks like:
![screen shot 2018-10-26 at 5 59 04 pm](https://user-images.githubusercontent.com/30536469/47594724-a131c180-d94a-11e8-9cc4-4cbba13f22ba.png)

Here is what  the documentation says that I should see:

![screen shot 2018-10-26 at 5 59 12 pm](https://user-images.githubusercontent.com/30536469/47594737-b149a100-d94a-11e8-917a-782ccc1fb3a4.png)

### Supporting Log Snippets
Also, no Logs are generated. My services just keep getting started and fail every ~30s.
![screen shot 2018-10-26 at 6 13 06 pm](https://user-images.githubusercontent.com/30536469/47594778-da6a3180-d94a-11e8-919a-5bfcaf86c11c.png)
![screen shot 2018-10-26 at 6 13 42 pm](https://user-images.githubusercontent.com/30536469/47594782-e524c680-d94a-11e8-8802-2433662469f4.png)

**Comment by adnxn** [comment]
> I am trying to run a basic Docker container on ECS, and I keep getting the error "STOPPED(Task failed to start)

just to clarify- is this a fargate task? 

and you're seeing `STOPPED(Task failed to start)` in the `Last status` column of the console?

**Comment by PiedPieper** [comment]
> just to clarify- is this a fargate task?
Yes, but I experienced the same issue using EC2.

> and you're seeing STOPPED(Task failed to start) in the Last status column of the console?
Yes, should see it in the images above unless you're referring to a different console

**Comment by PiedPieper** [comment]
Futhermore, the logs are empty 
![screen shot 2018-10-29 at 6 15 03 pm](https://user-images.githubusercontent.com/30536469/47683654-970bff00-dba6-11e8-880b-1f346838eeee.png)

**Comment by tsmirnov** [comment]
I have exactly the same issue. What is interesting, that I can launch the same task manually without a service. This problem only occurs, when a service tries to run a task.

**Comment by Aarushi2016** [comment]
Hi there - I am very new to Fargate deployment(had experience on ECS with EC2). 

Deployed a service using Fargate, but the task remains in PENDING status and unable to come into running status.

Task details shows
Stopped reason: Task failed to start

Container Details shows:
Status reason: DockerTimeoutError: Could not transition to started; timed out after waiting 3m0s

fyi...I did define, AssignPublicIp: 'ENABLED'
Could someone please advise on, what could be the possible reason for this error and how to find the root cause for the above  error.
Also, please advise, how do we look for errors in container as we see the container logs in EC2 method of deployments.

Really appreciate your help and time here!!

Regards,
Ravi

**Comment by calvinzhuca** [comment]
> Hi there - I am very new to Fargate deployment(had experience on ECS with EC2).
> 
> Deployed a service using Fargate, but the task remains in PENDING status and unable to come into running status.
> 
> Task details shows
> Stopped reason: Task failed to start
> 
> Container Details shows:
> Status reason: DockerTimeoutError: Could not transition to started; timed out after waiting 3m0s
> 
> fyi...I did define, AssignPublicIp: 'ENABLED'
> Could someone please advise on, what could be the possible reason for this error and how to find the root cause for the above error.
> Also, please advise, how do we look for errors in container as we see the container logs in EC2 method of deployments.
> 
> Really appreciate your help and time here!!
> 
> Regards,
> Ravi

Ravi, 

I suspect u might have the same issue as I do. I'm also just start using the fargate, and run into the exact same error "DockerTimeoutError: Could not transition to started; timed out after waiting 3m0s". 

After digging a while, I found it's the wrong cloudwatch log preventing my docker image from starting. So I change the container log setting from cloudwatch to default (means no log at all) and then the contain can be started by fargate services. 

calvin

**Comment by erikpb** [comment]
Had the same issue running Fargate in a private subnet without a NAT Gateway. Besides adding ECR, S3 and ECS endpoints as very well explained in this [article](https://aws.amazon.com/blogs/compute/setting-up-aws-privatelink-for-amazon-ecs-and-amazon-ecr/), adding an endpoint for AWS logs did the trick.

**Comment by ncipollina** [comment]
> Had the same issue running Fargate in a private subnet without a NAT Gateway. Besides adding ECR, S3 and ECS endpoints as very well explained in this [article](https://aws.amazon.com/blogs/compute/setting-up-aws-privatelink-for-amazon-ecs-and-amazon-ecr/), adding an endpoint for AWS logs did the trick.

How did you add an endpoint for AWS logs?

**Comment by calvinzhuca** [comment]
click on VPC --> endpoint, 
https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/cloudwatch-logs-and-interface-VPC.html

**Comment by erikpb** [comment]
Thanks @calvinzhuca for pre-empting my response ;-)

**Comment by Tarvinder91** [comment]
> 
> 
> Had the same issue running Fargate in a private subnet without a NAT Gateway. Besides adding ECR, S3 and ECS endpoints as very well explained in this [article](https://aws.amazon.com/blogs/compute/setting-up-aws-privatelink-for-amazon-ecs-and-amazon-ecr/), adding an endpoint for AWS logs did the trick.

Yes. If using Fargate launch type, you need these three endpoints : dkr.ecr, s3 and logs endpoint. ECS endpoint is not needed.

**Comment by venu-ibex-9** [comment]
Hi Guys,

I am using AWS ECS EC2 service and we are using Jenkins for CI/CD, when I update the latest task on the service, via Jenkins it showing me as a success but it's not running with the latest task it's failed to run with that. So here I need to notify this Via Jwwnkins Job, anyone can me on this please?
